internal/wasm/premissioncache.go

```diff
@@ -125,6 +128,7 @@ func newPermissionCache(options permissionCacheOptions, logger logging.Logger) *
 		cache:                  make(map[string]*permissionCacheEntry),
 		permissionCacheOptions: options,
 		logger:                 logger,
+		triggerChan:            make(chan struct{}),
```

Comment:
can this be avoided by using a really small checkInterval in tests ?

Reply:
Probably, but no guarantee as it's inherently flaky.

Reply:
can we use `waitUntil` in test?

Reply:
> can we use waitUntil in test?
Unlike HTTP Get, these tests are stateful, so `waitUntil` would be tricky.

Reply:
@arkodg you prefer adjusting the checkInterval ?

Reply:
yes prefer if we didn't have to add another field which is relevant only for tests, if possible